File: medjee/backend/app/models/paper.py
from sqlalchemy import Column, DateTime, Integer, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
import uuid

from app.db.base import Base


class Paper(Base):

    __tablename__ = "papers"

    id = Column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4
    )

    filename = Column(
        String(255),
        nullable=False
    )

    storage_path = Column(Text)

    std = Column(String(10))

    paper_type = Column(String(50))

    total_marks = Column(Integer)

    duration_mins = Column(Integer)

    uploaded_at = Column(
        DateTime(timezone=True),
        server_default=func.now()
    )

File: medjee/frontend/src/types/paper.ts
export type Paper = {
  id: string;

  filename: string;

  std?: string;

  paper_type?: string;

  total_marks?: number;

  duration_mins?: number;

  uploaded_at?: string;

  question_count?: number;

  subjects_count?: number;

  processing_status?: string;
};